docs/maven-tutorial.md

```diff
@@ -0,0 +1,218 @@
+# OpenShift + Maven Tutorial
+
+While developing your application with Java, you'll most likely be building
+many, many, many times.  In order to shorten the build times of your pods,
+dependencies can be cached in local Maven repositories.  This tutorial will step
```

Comment:
during the assemble script execution, a .m2 dir is created within the container.

so when the container is committed, that .m2 directory is still present in the image that results from the commit.

that means on future builds you can perform what is called an "incremental" build in which we pull down the previous output image, extract content from it (in this case, the .m2 directory), and inject it into the build process along with the source, so we don't have to re-download the dependencies from a remote maven repo.
https://github.com/openshift/source-to-image/blob/master/docs/sti-flow.png